app/controllers/spree/admin/payment_methods_controller.rb

```diff
@@ -40,8 +40,8 @@ def update
           @payment_method = PaymentMethod.find(params[:id])
         end
 
-        payment_method_params = params[ActiveModel::Naming.param_key(@payment_method)] || {}
-        attributes = params[:payment_method].merge(payment_method_params)
+        update_params = params[ActiveModel::Naming.param_key(@payment_method)] || {}
+        attributes = payment_method_params.merge(update_params)
```

Comment:
It's not very clear what's happening here, or why...? Can it be extracted a bit so it's easier to read?

Reply:
yes I can. it's spree code :-)

Reply:
ok, done in new commit.